plugin/src/test/java/dev/regadas/trino/pubsub/listener/PubSubEventListenerTest.java

```diff
@@ -155,7 +155,7 @@ void testQueryCreatedPublishCorrespondingMessage() {
 
         eventListener.queryCreated(TestData.FULL_QUERY_CREATED_EVENT);
 
-        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryCreatedEvent.class));
+        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryEvent.class));
```

Comment:
```suggestion
        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryEvent.class));
        assertThat(((Schema.QueryEvent) publisher.lastPublishedMessage).hasQueryCreatedEvent(), is(true);
```
Let's assert that the event wrapped into a QueryCreatedEvent is a query created event

The same with the rest

Reply:
Hi @narape , the above build failed, it there a cleaner way to rewrite this?
```
        try {
            var event = Schema.QueryEvent.parseFrom(publisher.lastPublishedMessage.toByteString());
            assertThat(event.hasQueryCreated(), is(true));
        } catch (Exception e) {
            fail(e.getMessage());
        }
```

Reply:
## generic solution

You could extract it into a function with a custom interface to properly model the `parseFrom` methods of the proto classes

```
    <T> T parseProto(Message lastPublishedMessage, ParsingFuntion<T> parseFunction) {
        try {
            return parseFunction.parse(lastPublishedMessage.toByteString());
        } catch (InvalidProtocolBufferException e) {
            fail(e);
            return null;
        }
    }
    
    interface ParsingFuntion<T> {
        T parse(ByteString bytes) throws InvalidProtocolBufferException;
    }
```

and then on the assertion you can use
```
        assertThat(parseProto(publisher.lastPublishedMessage, Schema.QueryEvent::parseFrom).hasQueryCreated(), is(true));
```

## simpler and more specific solution

```
    <T> T parseQueryEvent(Message lastPublishedMessage) {
        try {
            return Schema.QueryEvent.parseFrom(lastPublishedMessage.toByteString());
        } catch (InvalidProtocolBufferException e) {
            fail(e);
            return null;
        }
    }
```

and then on the assertion you can use
```
        assertThat(parseQueryEvent(publisher.lastPublishedMessage).hasQueryCreated(), is(true));
```